redaxo/src/core/lib/util/editor.php

```diff
@@ -38,28 +38,18 @@ public static function factory()
 
     public function getUrl($filePath, $line)
     {
-        // make internal urls work with parse_url()
-        $filePath = str_replace('rex:///', 'rex://', $filePath);
-
-        $parsedUrl = parse_url($filePath);
-        // rex:// internal url mapping to backend form-edit urls
-        if (isset($parsedUrl['scheme']) && $parsedUrl['scheme'] === 'rex') {
-            if ($parsedUrl['host'] === 'template') {
-                $templateId = ltrim($parsedUrl['path'], '/');
-                return rex_url::backendPage('templates', ['function' => 'edit', 'template_id' => $templateId]);
-            }
-            if ($parsedUrl['host'] === 'module') {
-                $moduleId = (int) ltrim($parsedUrl['path'], '/');
-                return rex_url::backendPage('modules/modules', ['function' => 'edit', 'module_id' => $moduleId]);
-            }
-        }
-
         $editor = rex::getProperty('editor');
         $editorUrl = $this->editors[$editor];
 
         $editorUrl = str_replace('%line', $line, $editorUrl);
         $editorUrl = str_replace('%file', $filePath, $editorUrl);
 
+        $editorUrl = rex_extension::registerPoint(new rex_extension_point('EDITOR_URL', $editorUrl, [
+            'file' => $filePath,
+            'line' => $line,
+            'editor' => $this,
```

Comment:
da wir den editor ja auch via factory bereitstellen, weiß ich nicht ob es sinnvoll ist diesen hier via parameter zu übergeben?

Reply:
Meine Überlegung war: Falls es mal, warum auch immer, doch irgendwie zwei unterschiedliche editor-Objekte gäbe (also mit unterschiedlichem Verhalten/Editor), wäre so sichergestellt, dass garantiert die URL innerhalb des EPs für den selben Editor erstellt wird.

Aber ist wahrscheinlich Quatsch, wird nie so sein. Von daher kann ich es gerne wieder entfernen.

Reply:
hmm so gesehen bräuchten wir eigentlich keine factory sondern eher einen singleton ;).

vllt sollten wir das system-editor object via rex::getProperty() rummreichen?
die factory ist ja eher dafür gedacht, dass man die eingentliche klasse zur laufzeit dann austauschen kann...

Reply:
aber vllt ist das auch zu viel wissenschaft und man nimmt nur die factory, da der use-case hier ja erstmal nicht gegeben ist...